ad7768-1fmcz/ad77681.c

```diff
@@ -243,10 +313,321 @@ int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
 
 	/* Fill the adc_data buffer */
 	memcpy(adc_data, buf, ARRAY_SIZE(buf));
+	
+	if (buf[1] & 0x80)
+		converted_data = (int32_t)(0xFF << 24 | (((buf[1]  &   0xFF)  << 16) |	
+												 ((buf[2]  &   0xFF)  << 8) |	
+												 ((buf[3]  &   0xFF)  << 0)));
+	else
+		converted_data = (int32_t)(0x00 << 24 | (((buf[1]  &   0xFF)  << 16) |	
+												 ((buf[2]  &   0xFF)  << 8) |	
+												 ((buf[3]  &   0xFF)  << 0)));
+	
+	Vref = ((double)(dev->vref)) / 1000.0;
+	*voltage = (double)(((2.0 * Vref) / AD7768_FULL_SCALE ) * converted_data);					// ((2*Vref)*code)/2^24	
+	
+	if ((*voltage > Vref) || (*voltage < Vref * (-1.0)))										// Overflow protection, voltage can only fit between +- Vref
+		ret = -1;
+
+	return ret;
+}
+
+/**
+ * This function uses continuous read mode, which has to be enabled in advance
+ * Read conversion result from device using GPIO DRDY interrupt in 16bit frames
+ *
+ *	    /-------------\       /-------------\       /-------------\
+ *------  24-bit data  -------  8-bit staus  -------   8-bit CRC   -------
+ *      \-------------/       \-------------/       \-------------/
+ *
+ *                THIS FUNCTION USES 16bit SPI FRAMES
+ *                
+ * Firtst 24bits or 16bits of data	dev->conv_length   followed by
+ * 8bits of status bit, if enabled	dev->sataus_bit    followed by
+ * 8bits of CRC, if enabled			dev->crc_sel
+ *
+ * @param dev - The device structure.
+ * @param measured_data - structure carying ADC measurement info
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_spi_read_interrupt_adc_data(struct ad77681_dev *dev,
+	struct adc_data *measured_data)
+{
+	int32_t ret;
+	uint16_t buf[3];
+	
+	buf[0] = 0;																					// 16bit buffers! Used only for ADC data read
+	buf[1] = 0;
+	buf[2] = 0;
+	
+	ret = spi_read_data(dev->spi_desc, buf, dev->data_frame_bytes);
+	if (ret < 0)
+		return ret;
+		
+	if (dev->conv_len == AD77681_CONV_24BIT)													// 24bit format
```

Comment:
please add comments like `// Cuts first 8 bits from buf[1] and paste it after 16bits of buf[0]` above the code that handles that specific comment. The number of characters per line should not be above 80 (only in exceptional situations).

check for similar issues throughout the file.